optimum/onnxruntime/trainer.py

```diff
@@ -279,12 +308,48 @@ def __init__(
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
         )
 
+        # We leverage both training_model and inference_model in conjunction with model.
+        # _training_model will be wrapped so it will use ORT and will use the overriden functions in ModuleWithLoss.
+        # _inferencing_model will be storing the default version of the model and we will switch to it in case of eval/test.
+
+        # Only Wrap the model if we pass --loss_in_train flag.
+        if args.loss_in_train:
+            self._training_model = ModuleWithLoss(model, args)
+        else:
+            self._training_model = model
```

Comment:
Do we still need to distinguish training model and inference model, if we have the module property with the wrapper, given that we unwrap self.model for inference here: https://github.com/huggingface/optimum/blob/c45bd53757f2b789139ff0076288d875940699ea/optimum/onnxruntime/trainer.py#L534